sdk/servicebus/service-bus/test/perf/azure-sb-package/send.ts

```diff
@@ -0,0 +1,125 @@
+/*
+# Overview
+Measures the maximum throughput of `sender.send()` in package `azure-sb`.
+
+# Instructions
+1. Create a Service Bus namespace with `Tier=Premium` and `Messaging Units=4`.  It is recommended to use the largest possible namespace to allow maximum client throughput.
+2. Create a queue inside the namespace.
+3. Set env vars `SERVICE_BUS_CONNECTION_STRING` and `SERVICE_BUS_QUEUE_NAME`.
+4. `ts-node send.ts [maxInflightMessages] [totalMessages]`
+5. Example: `ts-node send.ts 1000 1000000`
+ */
+
+import { createServiceBusService, ServiceBusService } from "azure-sb";
+import delay from "delay";
+import moment from "moment";
+
+const _payload = Buffer.alloc(1024);
+const _start = moment();
+
+let _sent = 0;
+let _accepted = 0;
+
+async function main(): Promise<void> {
+  // Endpoint=sb://<your-namespace>.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<shared-access-key>
+  const connectionString = process.env.SERVICE_BUS_CONNECTION_STRING as string;
+  const entityPath = process.env.SERVICE_BUS_QUEUE_NAME as string;
+
+  const maxInflight = process.argv.length > 2 ? parseInt(process.argv[2]) : 1;
+  const messages = process.argv.length > 3 ? parseInt(process.argv[3]) : 10;
+  log(`Maximum inflight messages: ${maxInflight}`);
+  log(`Total messages: ${messages}`);
+
+  const writeResultsPromise = WriteResults(messages);
+
+  await RunTest(connectionString, entityPath, maxInflight, messages);
+
+  await writeResultsPromise;
+}
+
+async function RunTest(
+  connectionString: string,
+  entityPath: string,
+  maxInflight: number,
+  messages: number
+): Promise<void> {
+  const sbService: ServiceBusService = createServiceBusService(connectionString);
+
+  function sendMessages(): void {
+    while (_sent < messages && inflight() < maxInflight) {
+      _sent++;
+      sbService.sendQueueMessage(entityPath, { body: JSON.stringify(_payload) }, function(
+        err: any
+      ) {
+        if (err) {
+          console.log(err.message);
```

Comment:
If a send operation fails, then `_sent` would be 1 more than `_accepted` at this point and the `while` loop in `WriteResults` wont exit

Reply:
The tests currently do not have any error handling, but I agree it would be good to add.  Assuming the errors are infrequent enough, it would be sufficient to just detect the error, print a message, and terminate the process.

Reply:
In that case, I would suggest to use another variable say `_rejected` and update the `while` condition in `WriteResults` to `_accepted + _rejected < messages`. Just to guard against hung test case

Reply:
Or use `_sent < messages` instead of `_accepted < messages`

Reply:
@ramya-rao-a 
Then, I guess, we'd have to update the rhea-promise sample as well.
https://github.com/Azure/azure-sdk-for-js/blob/2654f7eb3f4a97e58b9dca3e73409d72b5807623/sdk/servicebus/service-bus/test/perf/rhea-promise/send.ts#L132
You want me update that in this PR ?

Reply:
In that case send a separate PR to fix in both places

Reply:
Okay